src/rt/aaA.d

```diff
@@ -142,16 +142,36 @@ private:
     {
         auto obuckets = buckets;
         buckets = allocBuckets(ndim);
+        auto newFirstUsed = cast(uint) ndim;
 
-        foreach (ref b; obuckets)
+        foreach (ref b; obuckets[firstUsed .. $])
             if (b.filled)
-                *findSlotInsert(b.hash) = b;
+            {
+                auto x = findSlotInsert(b.hash);
+                newFirstUsed = min(newFirstUsed, cast(uint) (x - buckets.ptr));
```

Comment:
I'll put it back to setting it to 0, because that change is orthogonal to this PR. However, no reason to ignore the cache when referencing the original array. Will update soon.